icu4j/main/core/src/test/java/com/ibm/icu/dev/test/segmenter/SegmentsTest.java

```diff
@@ -104,22 +106,23 @@ public void testIsBoundary() {
 
     // Create new Segments for source1
     Segments segments1 = enWordSegmenter.segment(source1);
+    List<Segment> segments = segments1.segments().collect(Collectors.toList());
 
-    Object[][] casesData = {
-        {"start of segment",                     4,                     true},
-        {"between start and limit of segment",   6,                     false},
-        {"limit of segment",                     9,                     true},
-        {"beginning of string",                  0,                     true},
-        {"end of string",                        source1.length(),      true},
-    };
-
-    for (Object[] caseDatum : casesData) {
-      String desc = (String) caseDatum[0];
-      int idx = (int) caseDatum[1];
-      boolean exp = (boolean) caseDatum[2];
-
-      assertThat(desc, segments1.isBoundary(idx) == exp);
+    BitSet actBoundaries = new BitSet(source1.length());
```

Comment:
`new BitSet(source1.length() + 1);`

+1 so that the length itself fits

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/BitSet.html#%3Cinit%3E(int)

Reply:
Done.